build/Build.Pack.cs

```diff
@@ -649,4 +649,58 @@ string GetMicrok8sIpAddress()
         { "linux-arm64", ("arm64", "arm64") },
         { "linux-arm", ("armhf", "armv7") }
     };
+
+    AbsolutePath InstallOctopusCli()
+    {
+        const string cliVersion = "2.11.0";
+
+        // Windows uses octopus.exe, everything else uses octopus
+        var cliName = EnvironmentInfo.IsWin ? "octopus.exe" : "octopus";
+
+        var unversionedCliFolder = TemporaryDirectory / "octopus-cli";
+        var cliFolder = unversionedCliFolder / cliVersion;
+        var cliPath = cliFolder / cliName;
+        if (cliPath.FileExists())
+        {
+            // Assume it has already been installed
+            return cliPath;
+        }
+
+        cliFolder.CreateDirectory();
+
+        var osId = true switch
+        {
+            _ when EnvironmentInfo.IsWin => "windows",
+            _ when EnvironmentInfo.IsOsx => "macOS",
+            _ when EnvironmentInfo.IsLinux => "linux",
+            _ => throw new NotSupportedException("Unsupported OS")
+        };
+
+        var archId = EnvironmentInfo.IsArm64
+            ? "arm64"
+            : "amd64";
+
+        var archiveExtension = EnvironmentInfo.IsWin ? "zip" : "tar.gz";
+
+        var downloadUri = $"https://github.com/OctopusDeploy/cli/releases/download/v{cliVersion}/octopus_{cliVersion}_{osId}_{archId}.{archiveExtension}";
+
+        var archiveDestination = unversionedCliFolder / $"octopus-cli-archive.{cliVersion}.{archiveExtension}";
+
+        // If the archive already exists, we will download it again
+        archiveDestination.DeleteFile();
+
+        Log.Information("Downloading Octopus CLI from {downloadUri}", downloadUri);
+        HttpTasks.HttpDownloadFile(downloadUri, archiveDestination);
+
+        archiveDestination.UncompressTo(cliFolder);
+
+        if (!EnvironmentInfo.IsWin)
+        {
+            // We need to make the file executable as Nuke doesn't do that for us
+            ProcessTasks.StartProcess("chmod", $"+x {cliPath}").WaitForExit();
+        }
+
+        Assert.FileExists(cliPath, "The Octopus CLI executable was not found after extracting the archive");
```

Comment:
Looks like this check should be done before we try to make the file executable.

Reply:
Sure. I think I kept the assert by the return as a last line of defence if things went wrong (we actually build on Windows, so `chmod` isn't needed but it meant I could easily test this path locally).

I'll update, because at least that will give a consistent assertion message in both cases.